wiring/src/spark_wiring_ble.cpp

```diff
@@ -345,52 +339,46 @@ const uint8_t* BleUuid::rawBytes() const {
 }
 
 String BleUuid::toString(bool stripped) const {
-    String string;
-    if (type() == BleUuidType::SHORT) {
-        char buf[BLE_SIG_UUID_16BIT_LEN * 2 + 1] = {};
-        uint16_t bigEndian = uuid_.uuid16 << 8 | uuid_.uuid16 >> 8;
-        bytes2hexbuf((uint8_t*)&bigEndian, 2, buf);
-        buf[BLE_SIG_UUID_16BIT_LEN * 2] = '\0';
-        string.concat(buf);
-    } else {
-        uint8_t temp[BLE_SIG_UUID_128BIT_LEN];
-        toBigEndian(temp);
-        char* cStr = (char*)malloc(BLE_SIG_UUID_128BIT_LEN * 2 + 5);
-        if (cStr) {
-            SCOPE_GUARD ({
-                free(cStr);
-            });
-            if (stripped) {
-                bytes2hexbuf(temp, BLE_SIG_UUID_128BIT_LEN, cStr);
-                cStr[BLE_SIG_UUID_128BIT_LEN * 2] = '\0';
-            } else {
-                uint8_t idx = 0;
-                bytes2hexbuf(&temp[idx], 4, &cStr[0]);
-                idx += 4;
-                cStr[idx * 2] = '-';
-                bytes2hexbuf(&temp[idx], 2, &cStr[idx * 2 + 1]);
-                idx += 2;
-                cStr[idx * 2 + 1] = '-';
-                bytes2hexbuf(&temp[idx], 2, &cStr[idx * 2 + 2]);
-                idx += 2;
-                cStr[idx * 2 + 2] = '-';
-                bytes2hexbuf(&temp[idx], 2, &cStr[idx * 2 + 3]);
-                idx += 2;
-                cStr[idx * 2 + 3] = '-';
-                bytes2hexbuf(&temp[idx], 6, &cStr[idx * 2 + 4]);
-                idx += 6;
-                cStr[idx * 2 + 4] = '\0';
-            }
-            string.concat(cStr);
-        }
-    }
-    return string;
+    char cStr[BLE_SIG_UUID_128BIT_LEN * 2 + 5];
+    size_t len = toString(cStr, sizeof(cStr));
+    return String(cStr, len);
 }
 
 size_t BleUuid::toString(char* buf, size_t len, bool stripped) const {
-    String string = toString(stripped);
-    len = std::min(len, string.length());
-    memcpy(buf, string.c_str(), len);
+    if (!buf || len == 0) {
+        return 0;
+    }
+    if (type() == BleUuidType::SHORT) {
+        char cStr[BLE_SIG_UUID_16BIT_LEN * 2] = {};
+        uint16_t bigEndian = uuid_.uuid16 << 8 | uuid_.uuid16 >> 8;
+        bytes2hexbuf((uint8_t*)&bigEndian, 2, cStr);
+        len = std::min(len, sizeof(cStr));
+        memcpy(buf, cStr, len);
+        return len;
+    }
+    uint8_t temp[BLE_SIG_UUID_128BIT_LEN];
+    toBigEndian(temp);
+    char cStr[BLE_SIG_UUID_128BIT_LEN * 2 + 4];
```

Comment:
Why not use `buf` directly?

Reply:
If the length of given buffer is an odd number, the` bytes2hexbuf()` may corrupt the RAM. So we need an intermediate buffer for safe operation.